File: feedback/src/main/java/com/kanetik/feedback/model/ContextData.java
package com.kanetik.feedback.model;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class ContextData implements Serializable {
    static final long serialVersionUID = 327L;

    String title;
    List<ContextDataItem> contextData;

    public ContextData(String title) {
        this.title = title;
        this.contextData = new ArrayList<>();
    }

    public void add(String key, Object value) {
        this.contextData.add(new ContextDataItem(key, value));
    }

    public String toHtml() {
        StringBuilder builder = new StringBuilder("<h3>" + this.title + "</h3>");
        for (ContextDataItem contextItem : contextData) {
            builder.append("<b>").append(contextItem.key).append("</b>: ").append(contextItem.value).append("<br>");
        }

        return builder.toString();
    }

    public String toString() {
        StringBuilder builder = new StringBuilder(this.title + "\n\n");
        for (ContextDataItem contextItem : contextData) {
            builder.append(contextItem.key).append(": ").append(contextItem.value).append("\n");
        }

        return builder.toString();
    }
}

File: feedback/src/main/java/com/kanetik/feedback/model/ContextData.kt
package com.kanetik.feedback.model

import androidx.annotation.Keep
import java.lang.StringBuilder
import kotlinx.serialization.*

@Keep
@Serializable
data class ContextData(val title: String? = "Context Data") {
    var contextData: MutableList<ContextDataItem> = mutableListOf()

    fun add(key: String, value: String) {
        contextData.add(ContextDataItem(key, value))
    }

    fun add(key: String, value: Int) {
        contextData.add(ContextDataItem(key, value))
    }

    override fun toString(): String {
        val builder = StringBuilder(title!!)
                .append("\n")
                .append("------------------------")
                .append("\n")

        contextData.onEach { item ->
            builder
                    .append((item.key))
                    .append(": ")
                    .append(item.getValue())
                    .append("\n")
        }

        return builder.toString()
    }
}